src/main/java/org/sagebionetworks/web/client/widget/entity/controller/ProvenanceEditorWidgetViewImpl.java

```diff
@@ -0,0 +1,168 @@
+package org.sagebionetworks.web.client.widget.entity.controller;
+
+import org.gwtbootstrap3.client.ui.Button;
+import org.gwtbootstrap3.client.ui.Modal;
+import org.gwtbootstrap3.client.ui.TextArea;
+import org.gwtbootstrap3.client.ui.TextBox;
+import org.sagebionetworks.web.client.DisplayUtils;
+
+import com.google.gwt.event.dom.client.ClickEvent;
+import com.google.gwt.event.dom.client.ClickHandler;
+import com.google.gwt.uibinder.client.UiBinder;
+import com.google.gwt.uibinder.client.UiField;
+import com.google.gwt.user.client.ui.Composite;
+import com.google.gwt.user.client.ui.IsWidget;
+import com.google.gwt.user.client.ui.SimplePanel;
+import com.google.gwt.user.client.ui.Widget;
+import com.google.inject.Inject;
+
+public class ProvenanceEditorWidgetViewImpl extends Composite implements ProvenanceEditorWidgetView {
+
+	String PROV_LINK = "https://www.synapse.org/#!Wiki:syn2305384/ENTITY/62865";
+	
+	public interface ProvenanceEditorWidgetViewImplUiBinder 
+			extends UiBinder<Widget, ProvenanceEditorWidgetViewImpl> {}
+	
+	@UiField
+	Modal modal;
+	
+	@UiField
+	SimplePanel synAlertPanel;
+	
+	@UiField
+	TextBox editNameField;
+	
+	@UiField
+	TextArea editDescriptionField;
+	
+	@UiField
+	SimplePanel usedListPanel;
+	
+	@UiField
+	SimplePanel executedListPanel;
+	
+	@UiField
+	Button saveButton;
+	
+	@UiField
+	Button cancelButton;
+	
+	@UiField
+	SimplePanel entityFinderPanel;
+	
+	@UiField
+	SimplePanel urlDialogPanel;
+	
+	@UiField
+	Button helpButton;
+
+	
+	
+	Widget widget;
+	Presenter presenter;
+	
+	@Inject
+	public ProvenanceEditorWidgetViewImpl(ProvenanceEditorWidgetViewImplUiBinder binder) {
+		widget = binder.createAndBindUi(this);
+		saveButton.addClickHandler(new ClickHandler() {
+			@Override
+			public void onClick(ClickEvent event) {
+				presenter.onSave();
+			}
+		});
+		cancelButton.addClickHandler(new ClickHandler() {
+			@Override
+			public void onClick(ClickEvent event) {
+				modal.hide();
+			}
+		});
+		
+		helpButton.addClickHandler(new ClickHandler() {
+			
+			@Override
+			public void onClick(ClickEvent event) {
+				DisplayUtils.newWindow(PROV_LINK, "", "");
+			}
+		});
+	}
+	
+	@Override
+	public Widget asWidget() {
+		return widget;
+	}
+	
+	@Override
+	public void setVisible(boolean isVisible) {
```

Comment:
Please remove this setVisible() and just use the show() hide() methods that you wrote below.